creusot-contracts/src/logic/ord.rs

```diff
@@ -232,3 +232,42 @@ impl<A: OrdLogic, B: OrdLogic> OrdLogic for (A, B) {
     #[logic]
     fn eq_cmp(_: Self, _: Self) {}
 }
+
+impl<T: OrdLogic> OrdLogic for Option<T> {
+    #[logic]
+    fn cmp_log(self, o: Self) -> Ordering {
+        match (self, o) {
+            (None, None) => Ordering::Equal,
+            (None, Some(_)) => Ordering::Less,
+            (Some(_), None) => Ordering::Greater,
+            (Some(x), Some(y)) => x.cmp_log(y),
+        }
+    }
+
+    #[logic]
+    fn cmp_le_log(_: Self, _: Self) {}
+
+    #[logic]
+    fn cmp_lt_log(_: Self, _: Self) {}
+
+    #[logic]
+    fn cmp_ge_log(_: Self, _: Self) {}
+
+    #[logic]
+    fn cmp_gt_log(_: Self, _: Self) {}
+
+    #[logic]
+    fn refl(_: Self) {}
+
+    #[logic]
+    fn trans(_: Self, _: Self, _: Self, _: Ordering) {}
+
+    #[logic]
+    fn antisym1(_: Self, _: Self) {}
+
+    #[logic]
+    fn antisym2(_: Self, _: Self) {}
+
+    #[logic]
+    fn eq_cmp(_: Self, _: Self) {}
```

Comment:
These are `#[law]`s, and should duplicate the specifications (I know, this is a bit heavy). The reason why this does not trigger an error is that we don't run Why3 over

Reply:
I see, does this also apply to the functions that I omitted since they were already specified in the `OrdLogic` trait?

Reply:
Nope: default implementation are like in Rust: they don't need to be repeated.

Reply:
51ea82a37ddce74a4f84c3466e53911cf58f8aea should address this.